winml/adapter/winml_adapter_dml.cpp

```diff
@@ -19,10 +19,31 @@
 namespace winmla = Windows::AI::MachineLearning::Adapter;
 
 #ifdef USE_DML
+
+static bool IsCurrentModuleInSystem32() {
+  wil::unique_hmodule current_module;
+  FAIL_FAST_IF(0 == GetModuleHandleEx(
+                        GET_MODULE_HANDLE_EX_FLAG_FROM_ADDRESS,
+                        (LPCTSTR)IsCurrentModuleInSystem32,
+                        &current_module));
+
+  std::string current_module_path;
+  current_module_path.reserve(MAX_PATH);
+  auto size_module_path = GetModuleFileNameA(current_module.get(), current_module_path.data(), MAX_PATH);
+  FAIL_FAST_IF(size_module_path == 0);
+
+  std::string system32_path;
+  system32_path.reserve(MAX_PATH);
+  auto size_system32_path = GetSystemDirectoryA(system32_path.data(), MAX_PATH);
+  FAIL_FAST_IF(size_system32_path == 0);
```

Comment:
3 Fail_Fasts. Can they ever fail in a reasonable use that we would want to recover from?

Reply:
No I think these failures are fundamental. I expect these to always pass.